android/src/com/frostwire/android/gui/fragments/BrowsePeerFragment.java

```diff
@@ -485,6 +688,114 @@ private void changeSelectedRadioButton(byte currentFileType, byte nextFileType)
             nextButton.setChecked(true);
             nextButton.callOnClick();
         }
+
+    }
+
+    public boolean inSpecialState() {
+        return fragmentState != ViewState.NORMAL;
+    }
+
+    public void endSpecialState() {
+        if (fragmentState == ViewState.CHECK) {
+            endCheckState();
+        } else if (fragmentState == ViewState.FILTERING) {
+            endFilteringState();
+        } else if (fragmentState == ViewState.FILTERED) {
+            endFilteredState();
+        }
+    }
+
+    public void endCheckState() {
+        fragmentState = beforeCheckState;
+        if (beforeCheckState == ViewState.FILTERED) {
+            ((MainActivity) getActivity()).swapDrawerForBack();
+        }
+        swipe.setOnSwipeListener(swipeListener);
+        adapter.setCheckState(false);
+        updateViewsToState();
+    }
+
+    public void enterCheckState() {
```

Comment:
these enter and end methods make me a bit uneasy, in some cases there are questions about the previous states, in some other there are no such questions. Perhaps these methods are doing more than they should, perhaps they are being called in more places than they should (therefore these inconsistent checks)